### ISSUE TITLE
Dancing headers

### ISSUE THREAD

**Issue by DarkJaguar91** [timehop/sticky-headers-recyclerview#51]
If you have items that update in the recycler view, the header item for the set that updates tends to dance around.
Basically the header items moves to just above the item that has been updated and then moves back up to the correct position.

**Comment by DarkJaguar91** [comment]
Some investigation into the code revealed that it is caused by the StickyRecyclerHeadersDecoration.hasStickyHeader check.
The view at RecyclerView position 0 is not necessarily the view at the top of the screen, thus when multiple views on the screen are told to update, the header jumps down the the updating item before the draw occurs again and they are moved back to the top of the screen.

**Comment by DarkJaguar91** [comment]
I have implemented a fix on my side for the issue. I changed the hasStickyHeader method to the following.

```java
  private boolean hasStickyHeader(int viewTopOffset, int paddingHeight, int indexInList) {
    if (viewTopOffset > paddingHeight || mAdapter.getHeaderId(indexInList) < 0) {
      return false;
    }

    return true;
  } 
```
Of course the viewTopOffset and paddingHeight will have to be adjusted for both orientation and direction.

**Comment by DarkJaguar91** [comment]
Please see this video for an example of the "Dance": https://app.box.com/s/qr764as6rnye2orcwx4a92zrkscfuije

**Comment by jacobtabak** [comment]
Thanks for the detailed report and the fix suggestion.  I'll see if we can merge in your fix without breaking anything.  Do you see any potential issues?

**Comment by DarkJaguar91** [comment]
Well. So far it has only been my testing done on the fix that is running in my companies application. It will be another day or so (very likely next Tuesday only) before the full QA testing will be done.

But I have tested the following so far:
* The header item displays while scrolling on on all elements in the list
* The header doesn't ghost (create duplicates) even with a variable size list (elements do not have fixed size - like variable height message bubbles)
* The header items switch between each other correctly
* And of course they don't "dance" in any situation my companies app can produce.

I will get back to you after the full QA testing has been complete.

**Comment by tikimcfee** [comment]
Hey there. I wanted to chime in and mention that I too am seeing this behavior. Everything else is working great so far. 

We are currently using this library with https://github.com/h6ah4i/android-advancedrecyclerview, and the updates that occur in the natural flow of that library (specifically, notifying the full dataset of a change for drags / swipes) causes the "dancing".

Any update as to how this has progressed on your end, DarkJaguar91? I see that you modified the hasStickyHeader() method, but I don't know what the context for those parameters are. Any information or assistance on how you generate those params would be greatly appreciated!

Thanks much for your time on this - and for the great tool!

**Comment by DarkJaguar91** [comment]
Has been fixed by the latest pull request that was merged "Dancing Headers"

**Comment by jacobtabak** [comment]
Addressed in #53, released in 0.4.1.  Thanks @DarkJaguar91 !
